### DISCUSSION TITLE
[RFC] Static Extract style

### DISCUSSION THREAD

**Discussion by zombieJ** [ant-design/ant-design#40985] [RFCs - archive]
# 📋 RFC

## Summary

Provide a lib like `@ant-design/static-style-extract` to support generate static css for SSR usage to generate raw css file for caching.

## Basic example

```tsx
import extractStyle from `@ant-design/static-style-extract`;

const cssText = extractStyle(); // :where(.css-bAMboOo).ant-btn ...
```

## Motivation

ref #39891

Currently most SSR solution using inline html extract which makes html content large for every page first load. It should be a way to save the network request like origin css file solution.

But since this is enhancement of specific case, just bundle in `antd.theme` will make package size increase. It's unnecessary.

## Detailed

This pkg will default import all antd component from the entry and filter useless component like `Grid` or `notification` which does not affect SSR case. Loop render all the component and then extract the style from cssinjs cache.

For customize theme. `extractStyle` should support renderProps:

```tsx
import extractStyle from `@ant-design/static-style-extract`;

const cssText = extractStyle(node => (
  <ConfigProvider theme={{ ... }}>
    {node}
  </ConfigProvider>
)); 
```

## Drawbacks

For developer still need additional effort to use this generated style as file

## Adoption strategy

New npm package.

[a comment by an automated account is not shown]

**Comment by qinzehua** [comment]
请问在next.js page模式下怎么使用的？

**Comment by DicksonPunTS** [comment]
there is the same issue for antd layout
